feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.spec.ts

```diff
@@ -190,4 +167,52 @@ describe('CheckoutPaymentTypeComponent', () => {
       <any>mockActivatedRoute
     );
   });
+
+  describe('UI spinner when changing payment type', () => {
+    it('should display spinner when user selects a new payment and response did not complete', () => {
+      (
+        component.changeSelectedPaymentTypeInProgress$ as BehaviorSubject<boolean>
+      ).next(true);
+
+      fixture.detectChanges();
+
+      expect(el.query(By.css('div.cx-spinner'))).toBeTruthy();
+    });
+
+    it('should NOT display spinner when user selects a new payment and response did not complete', () => {
+      (
+        component.changeSelectedPaymentTypeInProgress$ as BehaviorSubject<boolean>
+      ).next(false);
```

Comment:
If the `changeSelectedPaymentTypeInProgress$` is false, then the user didn't select a payment type, did they?
If so, the title of the test is a bit confusing. Maybe we can just say: `should NOT display spinner when the payment type is NOT loading`?